src/Traits/FactoryTrait.php

```diff
@@ -154,24 +155,27 @@ public static function create(
         ?int $hour = null,
         ?int $minute = null,
         ?int $second = null,
+        ?int $microsecond = null,
```

Comment:
For users that already pass a timezone, this will throw a type error. $tz will never be an integer and if $microsecond isn't an integer then the type check fails.